source/common/http/codec_helper.h

```diff
@@ -25,6 +34,7 @@ class StreamCallbackHelper {
       return;
     }
 
+    reset_callbacks_started_ = true;
```

Comment:
Is it possible to just set reset_callbacks_run_ here and collapse into a single variable for simplicity? (Doesn't matter to me if you want to rename to reset_callbacks_started_). I think we don't have logic that requires the check to happen at the end?